streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -45,15 +45,24 @@
     private long maxSegmentId = -1L;
 
     Segments(final String name, final long retentionPeriod, final int numSegments) {
+        if (numSegments < 1) {
+            throw new IllegalArgumentException("numSegments must be positive");
+        }
+        if (retentionPeriod < 1) {
+            throw new IllegalArgumentException("retentionPeriod must be positive");
+        }
         this.name = name;
         this.numSegments = numSegments;
-        this.segmentInterval = Math.max(retentionPeriod / (numSegments - 1), MIN_SEGMENT_INTERVAL);
+        this.segmentInterval = numSegments == 1 ? Long.MAX_VALUE : Math.max(retentionPeriod / (numSegments - 1), MIN_SEGMENT_INTERVAL);
```

Comment:
Another thought while reading this line: maybe we should just forbid setting num.segments to 1 since otherwise we will not able to expire any windows at all, like the way we did here?